sdk/identity/azure-identity/azure/identity/_credentials/user_password.py

```diff
@@ -46,7 +42,7 @@ def __init__(self, client_id, username, password, **kwargs):
         # first time it's asked for a token. However, we want to ensure this first authentication is not silent, to
         # validate the given password. This class therefore doesn't document the authentication_record argument, and we
         # discard it here.
-        kwargs.pop("authentication_record", None)
+        kwargs.pop("_authentication_record", None)
```

Comment:
We don't want to honor "_authentication_record". why not just remove this line so we ignore "authentication_record"?

Reply:
I like leaving the reading from kwargs. It lowers the code churn and makes it less error prone to reintroduce support for these kwargs

Reply:
1. We can comment this line if pylint is not angry.
2. We can keep this line, it does not hurt.
3. The only possible side effect to rename it is if user uses "_authentication_record", they will get broken when we re-enable the feature.  - not too bad, but not good.

Reply:
1. Commenting the line breaks tests. We can skip them but then we may not catch regressions that complicate reverting these changes.
2. If we keep this and similar lines, code written against 1.4.0b7 will continue to work in 1.4.0 despite using unsupported features.
3. "_authentication_record" is underscored and undocumented: clearly not part of the public API

Reply:
I am confused.

If user passes in "authentication_record", do we want it break or not?

#1 will break it while #2 will not.

Seems to me you like neither of them. :)

Reply:
I'm sorry to have made confusion. I was responding to each of your points, not laying out a plan.

> If user passes in "authentication_record", do we want it break or not?
> 

We want to release 1.4.0 without the application authentication and cache configuration APIs present in 1.4.0b7. Not breaking code using these APIs is tantamount to including them in the stable release. So, we want it to break.

Reply:
Thanks for the clarification.

So I think just commenting this line will be good enough, right?

The only difference for adding "kwargs.pop("_authentication_record", None)" is if user passes "_authentication_record" (which is not a supported scenario), it will work in this release but will be broken when we bring the feature back. I don't think it is our intension to make this happen.

Reply:
> So I think just commenting this line will be good enough, right?
> 

Good enough for breaking the code we want to break but at the cost of breaking our tests. Skipping those would introduce a risk of unintentionally breaking features we want in our next beta. Also, the specific case you commented on here is a simple one because the argument is discarded. In other cases commenting a single line isn't enough because the argument is used.

> The only difference for adding "kwargs.pop("_authentication_record", None)" is if user passes "_authentication_record" (which is not a supported scenario), it will work in this release but will be broken when we bring the feature back. I don't think it is our intension to make this happen.

We don't support unsupported features.